context.go

```diff
@@ -1,16 +1,51 @@
 package basictracer
 
-// Context holds the basic Span metadata.
-type Context struct {
+import (
+	"sync"
+
+	"github.com/opentracing/opentracing-go"
+)
+
+// SpanContext holds the basic Span metadata.
+type SpanContext struct {
 	// A probabilistically unique identifier for a [multi-span] trace.
 	TraceID uint64
 
 	// A probabilistically unique identifier for a span.
 	SpanID uint64
 
-	// The SpanID of this Context's parent, or 0 if there is no parent.
-	ParentSpanID uint64
-
 	// Whether the trace is sampled.
 	Sampled bool
+
+	// The span's associated baggage.
+	baggageLock sync.Mutex
+	Baggage     map[string]string // initialized on first use
+}
+
+// BaggageItem belongs to the opentracing.SpanContext interface
+func (c *SpanContext) BaggageItem(key string) string {
+	// TODO: if we want to support onBaggage, need a pointer to the bt.Span.
+	//   s.onBaggage(canonicalKey, val)
+	//   if s.trim() {
+	//   	return s
+	//   }
+
+	c.baggageLock.Lock()
+	defer c.baggageLock.Unlock()
+
+	if c.Baggage == nil {
```

Comment:
@dominikh noted, though I still find this more readable / self-documenting re late-bound `c.Baggage` initialization and would like to keep it this way for that reason.